sdk/python/approzium/psycopg2.py

```diff
@@ -0,0 +1,107 @@
+import psycopg2
+import select
+import socket
+import struct
+import logging
+from ctypes import cdll, create_string_buffer
+from ctypes.util import find_library
+from .socketfromfd import fromfd
+from .authenticator import get_hash
+from .misc import read_int32_from_bytes
+
+libpq = cdll.LoadLibrary("libpq.so.5")
+libssl = cdll.LoadLibrary(find_library("ssl"))
+
+
+def advance_connection(pgconn):
+    state = pgconn.poll()
+    status = libpq.PQstatus(pgconn.pgconn_ptr)
+    fd = pgconn.fileno()
+    if state == psycopg2.extensions.POLL_OK:
+        pass
+    elif state == psycopg2.extensions.POLL_WRITE:
+        select.select([], [fd], [])
+    elif state == psycopg2.extensions.POLL_READ:
+        select.select([fd], [], [])
+    else:
+        raise psycopg2.OperationalError("poll() returned %s" % state)
+    return state, status
+
+
+def advance_until_challenge(pgconn):
+    CONNECTION_AWAITING_RESPONSE = 4
+    while True:
+        state, status = advance_connection(pgconn)
+        if status == CONNECTION_AWAITING_RESPONSE:
+            nbytes = 8096
```

Comment:
from an uninitiated reader's perspective, this choice for number of bytes seems arbitrary -- for instances like this perhaps a comment would be help understand what unit of data we're trying to read.